Mage.Sets/src/mage/cards/w/WordOfCommand.java

```diff
@@ -0,0 +1,249 @@
+    public WordOfCommandEffect() {
+        super(Outcome.GainControl);
+        this.staticText = "Look at target opponent's hand and choose a card from it. You control that player until Word of Command finishes resolving. The player plays that card if able. While doing so, the player can activate mana abilities only if they're from lands that player controls and only if mana they produce is spent to activate other mana abilities of lands the player controls and/or to play that card. If the chosen card is cast as a spell, you control the player while that spell is resolving";
+    }
+
+    public WordOfCommandEffect(final WordOfCommandEffect effect) {
+        super(effect);
+    }
+
+    @Override
+    public WordOfCommandEffect copy() {
+        return new WordOfCommandEffect(this);
+    }
+
+    @Override
+    public boolean apply(Game game, Ability source) {
+        Player controller = game.getPlayer(source.getControllerId());
+        Player targetPlayer = game.getPlayer(source.getFirstTarget());
+        MageObject sourceObject = game.getObject(source.getSourceId());
+        Card card = null;
+        if (controller != null && targetPlayer != null && sourceObject != null) {
+
+            // Look at target opponent's hand and choose a card from it
+            TargetCard targetCard = new TargetCard(Zone.HAND, new FilterCard());
+            if (controller.choose(Outcome.Discard, targetPlayer.getHand(), targetCard, game)) {
+                card = game.getCard(targetCard.getFirstTarget());
+            }
+
+            // You control that player until Word of Command finishes resolving
+            controller.controlPlayersTurn(game, targetPlayer.getId());
+            while (controller != null && controller.canRespond()) {
+                if (controller.chooseUse(Outcome.Benefit, "Resolve " + sourceObject.getLogName() + " now" + (card != null ? " and play " + card.getLogName() : "") + '?', source, game)) {
+                    // this is used to give the controller a little space to utilize his player controlling effect (look at face down creatures, hand, etc.)
+                    break;
+                }
+            }
+
+            // The player plays that card if able
+            if (card != null) {
+                // While doing so, the player can activate mana abilities only if they're from lands that player controls
+                RestrictionEffect effect = new WordOfCommandCantActivateEffect();
+                effect.setTargetPointer(new FixedTarget(targetPlayer.getId()));
+                game.addEffect(effect, source);
+
+                // and only if mana they produce is spent to activate other mana abilities of lands he or she controls and/or play that card
+                ManaPool manaPool = targetPlayer.getManaPool();
+                boolean autoPayment = manaPool.isAutoPayment();
+                boolean autoPaymentRestricted = manaPool.isAutoPaymentRestricted();
+                manaPool.setAutoPayment(true);
```

Comment:
later on in the code [here](https://github.com/magefree/mage/pull/4993/files#diff-d4b0cc23326c8768fc51d883f12fd221R472), inside the setAutoPayment() and setAutoPaymenRestricted() functions, we check if setForcedToPay is true and only if it is false do we actually set the values.

Is setting the auto payment options here actually needed? If we set forcedToPay earlier they would be superfluous

Reply:
I found it necessary to temporarily lock setAutoPayment and setAutoPaymentRestricted, because changing them isn't dependant on priority, and therefore otherwise the affected player could theoretically disable them manually before the effect is resolved, messing up the whole process.